### ISSUE TITLE
Can't create VM from ACG image to other resource group in Azure CLI 2.43.0

### ISSUE THREAD

**Issue by yoshit333** [Azure/azure-cli#24896]
## Describe the bug
When I create the VM from ACG (Azure Compute Gallery) image version, I faced the error  of "ResourceNotFound" even though resource group existing by using Azure CLI 2.43.0.
As I investigated this error can be reproduce when the resource group of VM and ACG are different.
If the resource group of VM and ACG is same, there is no isseu and could be deploy the VM.

**Command Name**
`az vm create`

**Errors:**
```
(ResourceNotFound) The Resource 'Microsoft.Compute/galleries/<RG>/images/<Image>' under resource group '<RG>' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
Code: ResourceNotFound
Message: The Resource 'Microsoft.Compute/galleries/<RG>/images/<Image>' under resource group '<RG>' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az vm create --resource-group {} --name {} --location {} --size {} --image {} --admin-username {} --admin-password {} --license-type {}`

Using Azure CLI 2.43.0.
Run the command as follows. 
az vm create --resource-group <RG1> --name <VM> --location <Region> --size Standard_D2s_v3 --image /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/<RG2>/providers/Microsoft.Compute/galleries/<ACG>/images/<Image>/versions/<x.x.x> --admin-username <username> --admin-password <password> --license-type Windows_Client

## Expected Behavior
Deploy the VM from ACG image version

## Environment Summary
```
Linux-5.4.0-1098-azure-x86_64-with-glibc2.35 (Cloud Shell), CBL-Mariner/Linux
Python 3.9.14
Installer: RPM

azure-cli 2.43.0

Extensions:
ai-examples 0.2.5
ml 2.11.0
ssh 1.1.3

Dependencies:
msal 1.20.0
azure-mgmt-resource 21.1.0b1
```
## Additional Context
As I checked, the workaround is using previous Azure CLI as 2.42.0.

**Comment by yonzhan** [comment]
@zhoxing-ms for awareness

**Comment by haxiel** [comment]
I have also encountered this issue today on Cloud Shell.

I enabled `--debug` for more information, and I could see that the command made one query using the correct resource ID of the image and received a valid response. However, it makes a second request later on, looking to find the same image under the resource group where the VM would be created. This attempt fails and and terminates the whole command.

**Comment by mcjhknauf** [comment]
Experienced the same issue, temporary solved using the next

```
sudo apt-get update
sudo apt-get install ca-certificates curl apt-transport-https lsb-release gnupg

curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/microsoft.gpg > /dev/null

AZ_REPO=$(lsb_release -cs)
echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | sudo tee /etc/apt/sources.list.d/azure-cli.list

sudo apt-get update
sudo apt-get install azure-cli=2.42.0-1~focal -y --allow-downgrades
```

**Comment by suresh-thelkar** [comment]
We are also observing the issue. Following are the steps followed.

- We have two subscriptions one for production and one for the dev and test purpose.
- Have created a new image version in ACG which resides in production subscription
- Trying to create a VM from a test subscription which actually has access to the above image version. Getting the following error.
`ERROR: (ResourceNotFound) The Resource 'Microsoft.Compute/galleries/...' under resource group 'xyz' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
Code: ResourceNotFound`

The command used to create a VM is given below.
`az vm create --resource-group <xyz> --image "image version resource id" --name "vm_name" --admin-user <abc> --generate-ssh-keys --security-type TrustedLaunch --enable-secure-boot true --enable-vtpm true`

We used to create a VM from a test subscription whereas the image version kept on the ACG is always in production subscription. This behavior was successfully working prior to az cli version "2.43.0"

For example I tried the same steps in az cli version "2.40.0". The same command works fine without any issues.

**Comment by yanzhudd** [comment]
> We are also observing the issue. Following are the steps followed.
> 
> * We have two subscriptions one for production and one for the dev and test purpose.
> * Have created a new image version in ACG which resides in production subscription
> * Trying to create a VM from a test subscription which actually has access to the above image version. Getting the following error.
>   `ERROR: (ResourceNotFound) The Resource 'Microsoft.Compute/galleries/...' under resource group 'xyz' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix Code: ResourceNotFound`
> 
> The command used to create a VM is given below. `az vm create --resource-group <xyz> --image "image version resource id" --name "vm_name" --admin-user <abc> --generate-ssh-keys --security-type TrustedLaunch --enable-secure-boot true --enable-vtpm true`
> 
> We used to create a VM from a test subscription whereas the image version kept on the ACG is always in production subscription. This behavior was successfully working prior to az cli version "2.43.0"
> 
> For example I tried the same steps in az cli version "2.40.0". The same command works fine without any issues.

Hi @suresh-thelkar, we have begun to fix this issue. May I ask if you could use the old version of the CLI for some time first? Our official release date is **2023-01-10**, and we recommend that this fix be released with the new version of CLI instead of a hotfix for some considerations:

- To ensure that fix is valid, the fix PR may take some time to be check validation and merged into main branch. But our release date is also coming soon.
- We were on vacation recently, so that if we release a hotfix for this bug, it is risk for the CLI stability.

We could first raise a PR and provide you a wheel to check whether the fix meet expectations. 
Could you please confirm if this solution is acceptable?

**Comment by yanzhudd** [comment]
Hi @suresh-thelkar, we have raised a PR for this bug, and the private package could be found in [Pipelines - Run 20221221.1 artifacts (azure.com)](https://dev.azure.com/azclitools/public/_build/results?buildId=21628&view=artifacts&pathAsName=false&type=publishedArtifacts). You could use this package to check if the fix meets expectations. Please let us know if there are any questions.

**Comment by suresh-thelkar** [comment]
> Hi @suresh-thelkar, we have begun to fix this issue. May I ask if you could use the old version of the CLI for some time first? Our official release date is **2023-01-10**, and we recommend that this fix be released with the new version of CLI instead of a hotfix for some considerations:
> 
> * To ensure that fix is valid, the fix PR may take some time to be check validation and merged into main branch. But our release date is also coming soon.
> * We were on vacation recently, so that if we release a hotfix for this bug, it is risk for the CLI stability.
> 
> We could first raise a PR and provide you a wheel to check whether the fix meet expectations. Could you please confirm if this solution is acceptable?

Thanks for the suggestion. In the interest of time and urgency we have downgraded az cli version and proceeding further.

**Comment by yanzhudd** [comment]
Hi @suresh-thelkar, this bug has been fixed in the new version of CLI (2.44.0). Please use `az upgrade` command to upgrade the version to apply this fix.

**Comment by suresh-thelkar** [comment]
> Hi @suresh-thelkar, this bug has been fixed in the new version of CLI (2.44.0). Please use `az upgrade` command to upgrade the version to apply this fix.

Thanks for the update. We will follow the suggested instructions accordingly and let you know in case if we run into any issues.
